### PR TITLE
chore: add browser testing infrastructure

### LINKED ISSUE #2612: Browser tests are inconsistently failing due to dynamic `istanbul` import
There are a few PRs currently [failing on CI](https://github.com/FuelLabs/fuels-ts/actions/runs/9649866949/job/26614887599?pr=2602#step:5:113) due to the failure to dynamically fetch [istanbul](https://github.com/gotwarlost/istanbul) for code coverage. This doesn't happen consistently though.

Here are a few affected PRs:

- https://github.com/FuelLabs/fuels-ts/pull/2600
- https://github.com/FuelLabs/fuels-ts/pull/2601
- https://github.com/FuelLabs/fuels-ts/pull/2602
- https://github.com/FuelLabs/fuels-ts/pull/2608

### PR DESCRIPTION
**Pull request by danielbate**
- Closes #2612

The majority of this work was introduced by @nedsalk in #2378, this PR will also resolve the flaky coverage import that is described in #2612. Investigation seems to point to flaky ESM bundling being the cause, which I believe to have been resolved [here](https://github.com/FuelLabs/fuels-ts/pull/2622#discussion_r1655305872).

### CODE REVIEW

File: vitest.browser.config.mts [hunk 1]
```diff
@@ -20,17 +20,25 @@ const config: UserConfig = {
...
       overrides: {
         fs: "memfs",
       },
     }),
   ],
   optimizeDeps: {
-    exclude: ["fsevents", "path-scurry"],
+    exclude: ["fsevents", "path-scurry", "@vitest/coverage-istanbul"],
     include: ["events", "timers/promises"],
+    entries: ["**/*.test.ts"],
```
Comment (danielbate) [inline vitest.browser.config.mts:35]: Most investigation seems to point to flaky ESM bundling. Haven't had any failed browser runs in 10 attempts after this change.

**Comment by danielbate** [comment]
@nedsalk any ideas on the validate changeset workflow? I've just reverted the revert commit, and introduced another changeset for my work.

**Comment by maschad** [comment]
> @nedsalk any ideas on the validate changeset workflow? I've just reverted the revert commit, and introduced another changeset for my work.

If the primary purpose of us using the changesets is to generate the changelogs [for our releases](https://github.com/FuelLabs/fuels-ts/pull/2553) , I think you can just consolidate [this changeset](https://github.com/FuelLabs/fuels-ts/pull/2622/files#diff-7f8db1fa064936a3c36a91c2df2f131e23eca3fa8f68ea1bbf6dd4ae1a3e5bf2) as well as [this changeset](https://github.com/FuelLabs/fuels-ts/pull/2622/files#diff-7f8db1fa064936a3c36a91c2df2f131e23eca3fa8f68ea1bbf6dd4ae1a3e5bf2),  into one changeset.

Then undo the deletion of [this changeset](https://github.com/FuelLabs/fuels-ts/pull/2622/files#diff-9a67f6a403afe07946435db59256a03c7cd53acbc2a892830cc92a145f4a26da)

> [!NOTE]
> Historically changesets have served a [different purpose](https://en.wikipedia.org/wiki/Changeset) to describe the exact differences between two successive versions of a change such as the git commit message, but based on our release workflow I don't think this applies.

**Comment by nedsalk** [comment]
@danielbate I suggest you rename your PR to be the same as #2378 and remove all the changesets except that PR's because this PR is reintroducing the same work as #2378. I'm suggesting this because #2378 won't show up in the changelog because it was reverted so there won't be any `chore: add browser testing infrastructure` entry in the changelog unless you do it with this PR.

**Review by nedsalk — DISMISSED** [review]

**Comment by danielbate** [comment]
Good points both 👍🏻 @nedsalk @maschad

**Review by nedsalk — APPROVED** [review]

File: .changeset/two-nails-report.md
Comment (nedsalk) [inline .changeset/two-nails-report.md:1]: Why is this changeset being deleted?

Comment (danielbate) [reply .changeset/two-nails-report.md:1]: 🤷🏻

Comment (danielbate) [reply .changeset/two-nails-report.md:1]: This is the changeset from the inital revert - https://github.com/FuelLabs/fuels-ts/pull/2618/files

**Review by maschad — APPROVED** [review]
Good find

**Review by Torres-ssf — APPROVED** [review]

**Pull request merged**